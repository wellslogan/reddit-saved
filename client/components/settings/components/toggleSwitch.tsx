import * as React from 'react';

type ToggleSwitchProps = {
  checked: boolean;
  onToggle: () => void;
  id: string;
};

export const ToggleSwitch: React.StatelessComponent<ToggleSwitchProps> = ({
  checked,
  onToggle,
  id,
}) => {
  return (
    <label className="switch">
      <input
        id={id}
        type="checkbox"
        checked={checked}
        onChange={() => onToggle()}
        data-testid={`toggleSwitch-${id}`}
      />
      <span className="slider round" />
    </label>
  );
};

ToggleSwitch.displayName = 'ToggleSwitch';
