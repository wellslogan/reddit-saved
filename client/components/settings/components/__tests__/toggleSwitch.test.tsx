import * as React from 'react';
import { render, fireEvent } from 'react-testing-library';

import { ToggleSwitch } from '../toggleSwitch';

describe('Component: ToggleSwitch', () => {
  it('should render checked', () => {
    const { container } = render(
      <ToggleSwitch id="test" checked={true} onToggle={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render unchecked', () => {
    const { container } = render(
      <ToggleSwitch id="test" checked={false} onToggle={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should call onToggle function on check', () => {
    const onToggle = jest.fn();
    const { getByTestId } = render(
      <ToggleSwitch id="test" checked={false} onToggle={onToggle} />
    );
    fireEvent.click(getByTestId('toggleSwitch-test'));
    expect(onToggle).toHaveBeenCalledTimes(1);
  });
});
