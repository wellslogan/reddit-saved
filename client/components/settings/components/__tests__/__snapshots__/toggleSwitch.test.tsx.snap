// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ToggleSwitch should render checked 1`] = `
<div>
  <label
    class="switch"
  >
    <input
      checked=""
      data-testid="toggleSwitch-test"
      id="test"
      type="checkbox"
    />
    <span
      class="slider round"
    />
  </label>
</div>
`;

exports[`Component: ToggleSwitch should render unchecked 1`] = `
<div>
  <label
    class="switch"
  >
    <input
      data-testid="toggleSwitch-test"
      id="test"
      type="checkbox"
    />
    <span
      class="slider round"
    />
  </label>
</div>
`;
